src/mesh/index_derivs.cxx

```diff
@@ -1211,7 +1211,7 @@ const Field3D Mesh::indexDDX(const Field3D &f, CELL_LOC outloc, DIFF_METHOD meth
   result = applyXdiff(f, func, diffloc);
   result.setLocation(diffloc); // Set the result location
 
-  result = interp_to(result, outloc); // Interpolate if necessary
+  result = interp_to(result, outloc, RGN_NOBNDRY); // Interpolate if necessary
```

Comment:
Why is this nobndry but others are noy/nox?

Reply:
I made it nobndry because I don't think the result needs valid guard cells. Inputs to derivatives do need guard cells at least in the direction of the derivative, hence the nox/noy.

Looking again, I think this interp_to couldn't ever be correct unless it's only interpolating in the z-direction, because the interpolation would need guard cells, so the result would have to be communicated and have a boundary condition applied. It might be better to raise an error for unhandled combinations of inloc/outloc instead?